### DISCUSSION TITLE
What's the difference in `mutate` from `useSWRConfig` vs `import`?

### DISCUSSION THREAD

**Discussion by ctsstc** [vercel/swr#1849] [Q&A]
I see the documentation using the hook `useSWRConfig`, it also explains the difference of a "bounded mutate", but I cannot seem to find anything about the mutate from the import.

Is there any difference aside from one being bound to the component lifecycle?
Using the import is quicker and more direct it seems.

## import

```jsx
import { mutate } from 'swr'
```

## useSWRConfig Hook

```jsx
{ mutate } = useSWRConfig()
```

**Comment by ctsstc** [comment]
Took some digging, the releases didn't explain much but the blog did: 
https://swr.vercel.app/blog/swr-v1#recommend-to-use-the-hook-returned-mutate

It sounds like you should use the `useSWRConfig` now. The `import` `mutate` hasn't been deprecated though [yet?]. It doesn't explain much of why this change was made. It does seem to allude towards saying that cache providers don't work with the `import` while `useSWRConfig` will work with cache providers.

**Reply by realyukii** [reply]
i am also curious too, is this information valid? ping @shuding @huozhi
